jsonschema/_utils.py

```diff
@@ -232,3 +248,148 @@ def uniq(container):
 
             seen.append(e)
     return True
+
+
+def find_evaluated_item_indexes_by_schema(validator, instance, schema):
+    """
+    Get all indexes of items that get evaluated under the current schema
+
+    Covers all keywords related to unevaluatedItems: items, prefixItems, if,
+    then, else, 'contains', 'unevaluatedItems', 'allOf', 'oneOf', 'anyOf'
+    """
+    if validator.is_type(schema, "boolean"):
+        return []
+    evaluated_indexes = []
+
+    if 'items' in schema:
+        return list(range(0, len(instance)))
+
+    if '$ref' in schema:
```

Comment:
I agree, this would also have a huge impact on performance and code reuse as `find_evaluated_item_indexes_by_schema` and `find_evaluated_property_keys_by_schema` are kind of repetitiv for certain tasks. I'm just afraid that this is a bigger change that has to be applied over all validator funtions somehow but let me know if you find a better approach :slightly_smiling_face: